optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/util/PromiseUtils.java

```diff
@@ -16,11 +16,60 @@
 
 package org.optaplanner.openshift.employeerostering.gwtui.client.util;
 
+import javax.inject.Singleton;
+
+import com.google.gwt.core.client.GWT;
+import elemental2.core.Error;
+import elemental2.core.JsObject;
 import elemental2.promise.Promise;
+import elemental2.promise.Promise.CatchOnRejectedCallbackFn;
+import elemental2.promise.Promise.PromiseExecutorCallbackFn;
 
+@Singleton
 public class PromiseUtils {
 
-    public static Promise<Void> resolve() {
+    public Promise<Void> resolve() {
         return Promise.resolve((Promise<Void>) null);
     }
+
+    public <T> Promise<T> promise(PromiseExecutorCallbackFn<T> executor) {
+        return manage(new Promise<>(executor));
+    }
+
+    public <T> Promise<T> manage(Promise<T> promise) {
+        return promise.catch_(this.<T> getDefaultCatch());
+    }
+
+    public <V> CatchOnRejectedCallbackFn<? extends V> getDefaultCatch() {
```

Comment:
I'd make this method private and expose another method called `invokeDefaultErrorHandling` or something with a potentially better name (😝) that does `promiseUtils.getDefaultCatch().onInvoke(i);`. Wdyt?

Reply:
Good idea, but that is more applicable to #93